File: distributions/1D_beta64_input.py
import numpy as np
from scipy.stats import beta

# Distribution name and prefix
prefix = "1D_beta64"
description = "Beta(6, 4)"

a = 6
b = 4

# Function to generate X values
def extract_samples(size):
    return np.random.beta(a, b, size=size)

# Function to calculate y values (PDF)
def compute_pdf(X):
    return beta.pdf(np.sort(X), a, b)

File: distributions/1D_beta64_input.R
# ============================================================
# 1D_beta64_input.R
# Beta(6, 4)
# ============================================================

extract_samples <- function(n) {
  return(matrix(rbeta(n, 6, 4), ncol = 1))
}

compute_pdf <- function(x) {
  dbeta(x, 6, 4)
}
